telegraf: release v1.25.3 and v1.26.0 (#4796)
influxdata/docs-v2: content/telegraf/v1.26/release-notes-changelog.md

### Features

- Always disable cgo support (static builds)
- Plugin state-persistence
- Add /etc/telegraf/telegraf.d to default config locations
- Print loaded configs
- Accept durations given in days (e.g. 7d)
- OAuth (`common.oauth`): Add audience parameter
- TLS (`common.tls`): Add enable flag
- CGroups (`inputs.cgroup`): Added support for cpu.stat
- Cisco Telemetry MDT (`inputs.cisco_telemetry_mdt`): Include delete field
- Disk (`inputs.disk`): Add label as tag
- DNS Query (`inputs.dns_query`): Add IP field(s)
- Docker Log (`inputs.docker_log`): Add state-persistence capabilities
- Ethtool (`inputs.ethtool`): Add support for link speed, duplex, etc.
- GNMI (`inputs.gnmi`): Set max gRPC message size
- HA Proxy (`inputs.haproxy`): Add support for tcp endpoints in haproxy plugin
- HTTP Listener v2 (`inputs.http_listener_v2`): Add custom server http headers
- Icinga2 (`inputs.icinga2`): Support collecting hosts, services, and endpoint metrics
- InfluxDB (`inputs.influxdb`): Collect uptime statistics
- Intel PowerStat (`inputs.intel_powerstat`): Add CPU base frequency metric and add support for new platforms
- Internet Speed (`inputs.internet_speed`):

>   - Add the best server selection via latency and jitter field
>   - Server ID include and exclude filter
> - JTI OpenConfig Telemtry (`inputs.jti_openconfig_telemetry`): Set timestamp from data
> - Modbus (`inputs.modbus`):
>   - Add RS485 specific config options
>   - Add workaround to enforce reads from zero for coil registers
>   - Allow to convert coil and discrete registers to boolean
> - MySQL (`inputs.mysql`): Add secret-store support
> - Open Weather Map (`inputs.openweathermap`): Add snow parameter
> - Processes (`inputs.processes`): Add use_sudo option for BSD
> - Prometheus (`inputs.prometheus`): Use namespace annotations to filter pods to be scraped
> - Redfish (`inputs.redfish`): Add power control metric
> - SQL Server (`inputs.sqlserver`): Get database pages performance counter
> - Stackdriver (`inputs.stackdriver`): Allow filtering by resource metadata labels
> - Statsd (`inputs.statsd`): Add pending messages stat and allow to configure number of threads
> - Vsphere (`inputs.vsphere`): Flag for more lenient behavior when connect fails on startup
> - Windows Event Log (`inputs.win_eventlog`): Add state-persistence capabilities
> - Windows Performance Counters (`inputs.win_perf_counters`): Add remote system support
> - Wireguard (`inputs.wireguard`): Add allowed_peer_cidr field
> - x509 Certificates (`inputs.x509_cert`):
>   - Add OCSP stapling information for leaf certificates (#10550)
>   - Add tag for certificate type-classification
> - MQTT (`outputs.mqtt`):
>   - Add option to specify topic layouts
>   - Add support for MQTT 5 publish properties
>   - Enhance routing capabilities
> - XPath Parser (`parsers.xpath`): Add timezone handling
> - Converter Processor (`processors.converter`): Convert tag or field as metric timestamp
> - Unpivot Processor (`processors.unpivot`): Add mode to create new metrics
> - Secret Stores:
>   - Add command-line option to specify password
>   - Add support for additional input plugins
>   - Convert many output plugins
> 
> ### Bugfixes
> 
> - Allow graceful shutdown on interrupt (e.g. Ctrl-C)
> - Only rotate log on SIGHUP if needed
> - AMQP Consumer (`inputs.amqp_consumer`):
>   - Avoid deprecations when handling defaults
>   - Fix panic on Stop() if not connected successfully
> - ethtool (`inputs.ethtool`): Close namespace file to prevent crash
> - statsd (`inputs.statsd`): On close, verify listener is not nil
> 
> ### Dependency Updates
> 
> - Update cloud.google.com/go/storage from 1.28.1 to 1.29.0
> - Update github.com/Azure/go-autorest/autorest/adal from 0.9.21 to 0.9.22
> - Update github.com/aliyun/alibaba-cloud-sdk-go from 1.62.77 to 1.62.193
> - Update github.com/aws/aws-sdk-go-v2/credentials from 1.13.2 to 1.13.15
> - Update github.com/aws/aws-sdk-go-v2/service/timestreamwrite from 1.14.5 to 1.16.0
> - Update github.com/coocood/freecache from 1.2.2 to 1.2.3
> - Update github.com/karrick/godirwalk from v1.17.0 to v1.16.2
> - Update github.com/opencontainers/runc from 1.1.3 to 1.1.4
> - Update github.com/opensearch-project/opensearch-go/v2 from 2.1.0 to 2.2.0
> - Update github.com/openzipkin-contrib/zipkin-go-opentracing from 0.4.5 to 0.5.0
> - Update github.com/rabbitmq/amqp091-go from 1.5.0 to 1.7.0
> - Update github.com/shirou/gopsutil from v3.22.12 to v3.23.2
> - Update github.com/stretchr/testify from 1.8.1 to 1.8.2
> - Update OpenTelemetry from 0.3.1 to 0.3.3

Reviewer: ```suggestion
## v1.26.0 [2023-03-13]

### Important Changes

- **Static builds**: Linux builds are now statically built. Other operating systems
  were cross-built in the past and as a result, already static. Users should
  not notice any change in behavior. The `_static` specific Linux binary is no
  longer produced as a result.
- **telegraf.d behavior**: The default behavior of reading
  `/etc/telegraf/telegraf.conf` now includes any `.conf` files under
  `/etc/telegraf/telegraf.d/`. This change will apply to the official Telegraf
  Docker image as well. This will simplify Docker usage when using multiple
  configuration files.
- **Default configuration**: The `telegraf config` command and default config file
  provided by Telegraf now includes all plugins and produces the same output
  across all operating systems. Plugin comments specify what platforms are
  supported or not.
- **State persistence**: State persistence is now available in select plugins. This
  will allow plugins to start collecting data, where they left off. A
  configuration with state persistence cannot change or it will not be able to
  recover.

### New Plugins

#### Inputs

- [Opensearch Query](https://github.com/influxdata/telegraf/tree/master/plugins/inputs/opensearch_query) (`inputs.opensearch_query`)
- [P4Runtime](https://github.com/influxdata/telegraf/tree/master/plugins/inputs/p4runtime) (`inputs.p4runtime`)
- [Radius Auth Response Time](https://github.com/influxdata/telegraf/tree/master/plugins/inputs/radius) (`inputs.radius`)
- [Windows Management Instrumentation (WMI)](https://github.com/influxdata/telegraf/tree/master/plugins/inputs/win_wmi) (`inputs.win_wmi`)

#### Parsers

- [Apache Avro](https://github.com/influxdata/telegraf/tree/master/plugins/parsers/avro) (`parsers.avro`)

#### Processors

- [lookup](https://github.com/influxdata/telegraf/tree/master/plugins/processors/lookup) (`processors.lookup`)

### Features

- Always disable cgo support (static builds).
- Plugin state-persistence.
- Add `/etc/telegraf/telegraf.d` to default configuration file locations.
- Print loaded configurationss.
- Accept durations given in days (e.g. 7d).
- OAuth (`common.oauth`): Add `audience` parameter.
- TLS (`common.tls`): Add `enable` flag.
- CGroups (`inputs.cgroup`): Add support for `cpu.stat`.
- Cisco Telemetry MDT (`inputs.cisco_telemetry_mdt`): Include `delete` field.
- Disk (`inputs.disk`): Add label as tag.
- DNS Query (`inputs.dns_query`): Add IP fields.
- Docker Log (`inputs.docker_log`): Add state-persistence capabilities.
- Ethtool (`inputs.ethtool`): Add support for link speed, duplex, etc.
- GNMI (`inputs.gnmi`): Set max gRPC message size.
- HA Proxy (`inputs.haproxy`): Add support for TCP endpoints in haproxy plugin.
- HTTP Listener v2 (`inputs.http_listener_v2`): Add custom server HTTP headers.
- Icinga2 (`inputs.icinga2`): Support collecting hosts, services, and endpoint metrics.
- InfluxDB (`inputs.influxdb`): Collect uptime statistics.
- Intel PowerStat (`inputs.intel_powerstat`): Add CPU base frequency metric and add support for new platforms.
- Internet Speed (`inputs.internet_speed`):
  - Add the best server selection via latency and jitter field.
  - Server ID include and exclude filter.
- JTI OpenConfig Telemtry (`inputs.jti_openconfig_telemetry`): Set timestamp from data.
- Modbus (`inputs.modbus`):
  - Add RS485 specific config options.
  - Add workaround to enforce reads from zero for coil registers.
  - Allow to convert coil and discrete registers to boolean.
- MySQL (`inputs.mysql`): Add secret-store support.
- Open Weather Map (`inputs.openweathermap`): Add `snow` parameter.
- Processes (`inputs.processes`): Add use_sudo option for BSD.
- Prometheus (`inputs.prometheus`): Use namespace annotations to filter pods to be scraped.
- Redfish (`inputs.redfish`): Add power control metric.
- SQL Server (`inputs.sqlserver`): Get database pages performance counter.
- Stackdriver (`inputs.stackdriver`): Allow filtering by resource metadata labels.
- Statsd (`inputs.statsd`): Add pending messages stat and allow to configure number of threads.
- Vsphere (`inputs.vsphere`): Flag for more lenient behavior when connect fails on startup.
- Windows Event Log (`inputs.win_eventlog`): Add state-persistence capabilities.
- Windows Performance Counters (`inputs.win_perf_counters`): Add remote system support.
- Wireguard (`inputs.wireguard`): Add allowed_peer_cidr field.
- x509 Certificates (`inputs.x509_cert`):
  - Add OCSP stapling information for leaf certificates.
  - Add tag for certificate type-classification.
- MQTT (`outputs.mqtt`):
  - Add option to specify topic layouts.
  - Add support for MQTT 5 publish properties.
  - Enhance routing capabilities.
- XPath Parser (`parsers.xpath`): Add timezone handling.
- Converter Processor (`processors.converter`): Convert tag or field as metric timestamp.
- Unpivot Processor (`processors.unpivot`): Add mode to create new metrics.
- Secret Stores:
  - Add command-line option to specify password.
  - Add support for additional input plugins.
  - Convert many output plugins.

### Bugfixes

- Allow graceful shutdown on interrupt (e.g. Ctrl-C).
- Only rotate log on SIGHUP if needed.
- AMQP Consumer (`inputs.amqp_consumer`):
  - Avoid deprecations when handling defaults.
  - Fix panic on Stop() if not connected successfully.
- ethtool (`inputs.ethtool`): Close namespace file to prevent crash.
- statsd (`inputs.statsd`): On close, verify listener is not nil.

### Dependency Updates

- Update cloud.google.com/go/storage from 1.28.1 to 1.29.0.
- Update github.com/Azure/go-autorest/autorest/adal from 0.9.21 to 0.9.22.
- Update github.com/aliyun/alibaba-cloud-sdk-go from 1.62.77 to 1.62.193.
- Update github.com/aws/aws-sdk-go-v2/credentials from 1.13.2 to 1.13.15.
- Update github.com/aws/aws-sdk-go-v2/service/timestreamwrite from 1.14.5 to 1.16.0.
- Update github.com/coocood/freecache from 1.2.2 to 1.2.3.
- Update github.com/karrick/godirwalk from v1.17.0 to v1.16.2.
- Update github.com/opencontainers/runc from 1.1.3 to 1.1.4.
- Update github.com/opensearch-project/opensearch-go/v2 from 2.1.0 to 2.2.0.
- Update github.com/openzipkin-contrib/zipkin-go-opentracing from 0.4.5 to 0.5.0.
- Update github.com/rabbitmq/amqp091-go from 1.5.0 to 1.7.0.
- Update github.com/shirou/gopsutil from v3.22.12 to v3.23.2.
- Update github.com/stretchr/testify from 1.8.1 to 1.8.2.
- Update OpenTelemetry from 0.3.1 to 0.3.3.
```

> ## v1.25.3 [2023-02-27]
> 
> ### Bugfixes
> 
> - Fix reload config on config update/SIGHUP
> - Bond (`inputs.bond`): Reset slave stats for each interface
> - Cloudwatch (`inputs.cloudwatch`): Verify endpoint is not nil
> - LVM (`inputs.lvm`): Add options to specify path to binaries
> - XPath (`parsers.xpath`): Fix panic for JSON name expansion
> - JSON (`serializers.json`): Fix stateful transformations
> 
> ### Dependency Updates
> 
> - Update cloud.google.com/go/pubsub from 1.27.1 to 1.28.0
> - Update github.com/containerd/containerd from 1.6.8 to 1.6.18
> - Update github.com/go-logfmt/logfmt from 0.5.1 to 0.6.0
> - Update github.com/gofrs/uuid from 4.3.1 to 5.0.0
> - Update github.com/gophercloud/gophercloud from 1.0.0 to 1.2.0
> - Update github.com/pion/dtls/v2 from 2.1.5 to 2.2.4
> - Update golang.org/x/net from 0.5.0 to 0.7.0
> - Update golang.org/x/sys from 0.4.0 to 0.5.0
> - Update google.golang.org/grpc from 1.52.3 to 1.53.0
> - Update k8s.io/apimachinery from 0.25.3 to 0.25.6
> - Update testcontainers from 0.14.0 to 0.18.0

Reviewer: ```suggestion
## v1.25.3 [2023-02-27]

### Bugfixes

- Fix reload config on config update/SIGHUP.
- Bond (`inputs.bond`): Reset slave stats for each interface.
- Cloudwatch (`inputs.cloudwatch`): Verify endpoint is not nil.
- LVM (`inputs.lvm`): Add options to specify path to binaries.
- XPath (`parsers.xpath`): Fix panic for JSON name expansion.
- JSON (`serializers.json`): Fix stateful transformations.

### Dependency Updates

- Update cloud.google.com/go/pubsub from 1.27.1 to 1.28.0.
- Update github.com/containerd/containerd from 1.6.8 to 1.6.18.
- Update github.com/go-logfmt/logfmt from 0.5.1 to 0.6.0.
- Update github.com/gofrs/uuid from 4.3.1 to 5.0.0.
- Update github.com/gophercloud/gophercloud from 1.0.0 to 1.2.0.
- Update github.com/pion/dtls/v2 from 2.1.5 to 2.2.4.
- Update golang.org/x/net from 0.5.0 to 0.7.0.
- Update golang.org/x/sys from 0.4.0 to 0.5.0.
- Update google.golang.org/grpc from 1.52.3 to 1.53.0.
- Update k8s.io/apimachinery from 0.25.3 to 0.25.6.
- Update testcontainers from 0.14.0 to 0.18.0.
```

## v1.25.2 [2023-02-13]

### Bug fixes
- Only read the config once
- fix link to license for Google flatbuffers
- Cisco Telemetry MDT (`inputs.cisco_telemetry_mdt`): Check subfield sizes to avoid panics
- Cloudwatch (`inputs.cloudwatch`): Enable custom endpoint support
- Conntrack (`inputs.conntrack`): Resolve segfault when setting collect field
- GNMI (`inputs.gnmi`): Handle both new-style tag_subscription and old-style tag_only
- MongoDB (`inputs.mongodb`):
  - Improve error logging
  - SIGSEGV when restarting MongoDB node
- MySQL (`inputs.mysql`): Avoid side-effects for TLS between plugin instances
- Prometheus (`inputs.prometheus`): Deprecate and rename the timeout variable
- Tail (`inputs.tail`): Fix typo in the README
- UPSD (`inputs.upsd`): Add additional fields
- x509 Cert (`inputs.x509_cert`): Fix Windows path handling
- Prometheus Client (`outputs.prometheus_client`): Expire with ticker, not add/collect
- Secret Stores: Check store id format and presence

### Dependency Updates

> - Update cloud.google.com/go/bigquery from 1.44.0 to 1.45.0
> - Update github.com/99designs/keyring from 1.2.1 to 1.2.2
> - Update github.com/antchfx/xmlquery from 1.3.12 to 1.3.15
> - Update github.com/antchfx/xpath from 1.2.2 to 1.2.3
> - Update github.com/coreos/go-semver from 0.3.0 to 0.3.1
> - Update github.com/moby/ipvs from 1.0.2 to 1.1.0
> - Update github.com/multiplay/go-ts3 from 1.0.1 to 1.1.0
> - Update github.com/prometheus/client_golang from 1.13.1 to 1.14.0
> - Update github.com/shirou/gopsutil from 3.22.9 to 3.22.12
> - Update go.mongodb.org/mongo-driver from 1.11.0 to 1.11.1
> - Update golang/x dependencies
> - Update google.golang.org/grpc from 1.51.0 to 1.52.0
> - Update google.golang.org/grpc from 1.52.0 to 1.52.3
> 
> ## v1.25.1 [2023-01-30]
> 
> ### Bug fixes
> - Catch non-existing commands and error out
> - Correctly reload configuration files
> - Handle float time with fractions of seconds correctly
> - Only set default snmp after reading all configs
> - Allow any 2xx status code
> - Kafka: Add keep-alive period setting for input and output
> - Cisco Telemetry MDT (`inputs.cisco_telemetry_mdt`): Add operation-metric and class-policy prefix
> - Exec (`inputs.exec`): Restore pre-v1.21 behavior for CSV data_format
> - GNMI (`inputs.gnmi`): Update configuration documentation
> - Logstash (`inputs.logstash`): Collect opensearch specific stats
> - MySQL (`inputs.mysql`): Revert slice declarations with non-zero initial length
> - OPC UA (`inputs.opcua`): Fix opcua and opcua-listener for servers using password-based auth
> - Prometheus (`inputs.prometheus`):
>   - Correctly track deleted pods
>   - Set the timeout for slow running API endpoints correctly
> - SQL Server (`inputs.sqlserver`):
>   - Add more precise version check
>   - Added own SPID filter
>   - SqlRequests include sleeping sessions with open transactions
>   - Suppress error on secondary replicas
> - UPSD (`inputs.upsd`):
>   - Always convert to float
>   - Ensure firmware is always a string
> - Windows Event Log (`inputs.win_eventlog`): Handle remote events more robustly
> - x509 Cert (`inputs.x509_cert`): Fix off-by-one when adding intermediate certificates
> - Loki (`outputs.loki`): Return response body on error
> - JSON v2 parser (`parsers.json_v2`): In case of invalid json, log message to debug log
> - Secret Stores:
>   - Cleanup duplicate printing
>   - Fix handling of "id" and print failing secret-store
>   - Fix handling of TOML strings
> 
> ### Dependency Updates
> - Update cloud.google.com/go/storage from 1.23.0 to 1.28.1
> - Update github.com/antchfx/jsonquery from 1.3.0 to 1.3.1
> - Update github.com/aws/aws-sdk-go-v2 from 1.17.1 to 1.17.3
> - Update github.com/aws/aws-sdk-go-v2/service/ec2 from 1.54.4 to 1.80.1
> - Update github.com/denisenkom/go-mssqldb from 0.12.0 to 0.12.3
> - Update github.com/eclipse/paho.mqtt.golang from 1.4.1 to 1.4.2
> - Update github.com/hashicorp/consul/api from 1.15.2 to 1.18.0
> - Update github.com/karrick/godirwalk from 1.16.1 to 1.17.0
> - Update github.com/kardianos/service from 1.2.1 to 1.2.2
> - Update github.com/nats-io/nats-server/v2 from 2.9.4 to 2.9.9

Reviewer: ```suggestion
## v1.25.2 [2023-02-13]

### Bug fixes
- Only read the config once.
- fix link to license for Google flatbuffers.
- Cisco Telemetry MDT (`inputs.cisco_telemetry_mdt`): Check subfield sizes to avoid panics.
- Cloudwatch (`inputs.cloudwatch`): Enable custom endpoint support.
- Conntrack (`inputs.conntrack`): Resolve segfault when setting collect field.
- GNMI (`inputs.gnmi`): Handle both new-style tag_subscription and old-style tag_only.
- MongoDB (`inputs.mongodb`):
  - Improve error logging.
  - SIGSEGV when restarting MongoDB node.
- MySQL (`inputs.mysql`): Avoid side-effects for TLS between plugin instances.
- Prometheus (`inputs.prometheus`): Deprecate and rename the timeout variable.
- Tail (`inputs.tail`): Fix typo in the README.
- UPSD (`inputs.upsd`): Add additional fields.
- x509 Cert (`inputs.x509_cert`): Fix Windows path handling.
- Prometheus Client (`outputs.prometheus_client`): Expire with ticker, not add/collect.
- Secret Stores: Check store id format and presence.

### Dependency Updates
- Update cloud.google.com/go/bigquery from 1.44.0 to 1.45.0.
- Update github.com/99designs/keyring from 1.2.1 to 1.2.2.
- Update github.com/antchfx/xmlquery from 1.3.12 to 1.3.15.
- Update github.com/antchfx/xpath from 1.2.2 to 1.2.3.
- Update github.com/coreos/go-semver from 0.3.0 to 0.3.1.
- Update github.com/moby/ipvs from 1.0.2 to 1.1.0.
- Update github.com/multiplay/go-ts3 from 1.0.1 to 1.1.0.
- Update github.com/prometheus/client_golang from 1.13.1 to 1.14.0.
- Update github.com/shirou/gopsutil from 3.22.9 to 3.22.12.
- Update go.mongodb.org/mongo-driver from 1.11.0 to 1.11.1.
- Update golang/x dependencies.
- Update google.golang.org/grpc from 1.51.0 to 1.52.0.
- Update google.golang.org/grpc from 1.52.0 to 1.52.3.

## v1.25.1 [2023-01-30]

### Bug fixes
- Catch non-existing commands and error out.
- Correctly reload configuration files.
- Handle float time with fractions of seconds correctly.
- Only set default snmp after reading all configs.
- Allow any 2xx status code.
- Kafka: Add keep-alive period setting for input and output.
- Cisco Telemetry MDT (`inputs.cisco_telemetry_mdt`): Add operation-metric and class-policy prefix.
- Exec (`inputs.exec`): Restore pre-v1.21 behavior for CSV data_format.
- GNMI (`inputs.gnmi`): Update configuration documentation.
- Logstash (`inputs.logstash`): Collect opensearch specific stats.
- MySQL (`inputs.mysql`): Revert slice declarations with non-zero initial length.
- OPC UA (`inputs.opcua`): Fix opcua and opcua-listener for servers using password-based auth.
- Prometheus (`inputs.prometheus`):
  - Correctly track deleted pods.
  - Set the timeout for slow running API endpoints correctly.
- SQL Server (`inputs.sqlserver`):
  - Add more precise version check.
  - Added own SPID filter.
  - SqlRequests include sleeping sessions with open transactions.
  - Suppress error on secondary replicas.
- UPSD (`inputs.upsd`):
  - Always convert to float.
  - Ensure firmware is always a string.
- Windows Event Log (`inputs.win_eventlog`): Handle remote events more robustly.
- x509 Cert (`inputs.x509_cert`): Fix off-by-one when adding intermediate certificates.
- Loki (`outputs.loki`): Return response body on error.
- JSON v2 parser (`parsers.json_v2`): In case of invalid json, log message to debug log.
- Secret Stores:
  - Cleanup duplicate printing.
  - Fix handling of "id" and print failing secret-store.
  - Fix handling of TOML strings.

### Dependency Updates
- Update cloud.google.com/go/storage from 1.23.0 to 1.28.1.
- Update github.com/antchfx/jsonquery from 1.3.0 to 1.3.1.
- Update github.com/aws/aws-sdk-go-v2 from 1.17.1 to 1.17.3.
- Update github.com/aws/aws-sdk-go-v2/service/ec2 from 1.54.4 to 1.80.1.
- Update github.com/denisenkom/go-mssqldb from 0.12.0 to 0.12.3.
- Update github.com/eclipse/paho.mqtt.golang from 1.4.1 to 1.4.2.
- Update github.com/hashicorp/consul/api from 1.15.2 to 1.18.0.
- Update github.com/karrick/godirwalk from 1.16.1 to 1.17.0.
- Update github.com/kardianos/service from 1.2.1 to 1.2.2.
- Update github.com/nats-io/nats-server/v2 from 2.9.4 to 2.9.9.
```
